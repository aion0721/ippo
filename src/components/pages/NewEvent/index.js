import React from 'react';
import { withRouter } from 'react-router-dom';
import styled from 'styled-components';
import propTypes from 'prop-types';
import NavigationBar from 'src/components/molecules/NavigationBar';
import EventCreateForm from 'src/components/organisms/EventCreateForm';

const Container = styled.div`
  padding: 10px 15px;
`;

const NewEvent = ({ createEvent, history }) => {
  return (
    <>
      <NavigationBar appName="IPPO" />
      <Container>
        <EventCreateForm
          onSubmit={event => {
            history.push('/events');
            return createEvent({ variables: { event } });
          }}
        />
      </Container>
    </>
  );
};

NewEvent.displayName = 'NewEvent';

NewEvent.propTypes = {
  createEvent: propTypes.func,
};

NewEvent.defaultProps = {};

export default withRouter(NewEvent);
