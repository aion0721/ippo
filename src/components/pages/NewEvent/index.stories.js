import React from 'react';
import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';
import NewEvent from '.';

const stories = storiesOf('pages/NewEvent', module);

stories.add('通常パターン', () => <NewEvent createEvent={action('createEvent')} />);
