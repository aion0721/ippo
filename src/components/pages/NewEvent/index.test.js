import React from 'react';
import { snapshot } from 'test/helpers';
import NewEvent from '.';

const props = {
  createEvent: jest.fn(),
};

snapshot('NewEvent/nomal', <NewEvent {...props} />);
