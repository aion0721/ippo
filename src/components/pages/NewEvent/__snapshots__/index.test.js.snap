// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot: NewEvent/nomal 1`] = `
.c0 {
  padding: 10px 15px;
}

<NewEvent
  createEvent={[MockFunction]}
>
  <NavigationBar
    appName="IPPO"
  >
    <Blueprint3.Navbar>
      <div
        className="bp3-navbar"
      >
        <Blueprint3.NavbarGroup
          align="left"
        >
          <div
            className="bp3-navbar-group bp3-align-left"
          >
            <Blueprint3.NavbarHeading>
              <div
                className="bp3-navbar-heading"
              >
                IPPO
              </div>
            </Blueprint3.NavbarHeading>
          </div>
        </Blueprint3.NavbarGroup>
      </div>
    </Blueprint3.Navbar>
  </NavigationBar>
  <styled.div>
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bdVaJa",
            "isStatic": true,
            "lastClassName": "c0",
            "rules": Array [
              "
  padding: 10px 15px;
",
            ],
          },
          "displayName": "styled.div",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bdVaJa",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
    >
      <div
        className="c0"
      >
        <EventCreateForm
          onSubmit={[Function]}
        >
          <form>
            <Blueprint3.FormGroup
              label="イベント名"
              labelFor="title"
              labelInfo="(必須)"
            >
              <div
                className="bp3-form-group"
              >
                <label
                  className="bp3-label"
                  htmlFor="title"
                >
                  イベント名
                   
                  <span
                    className="bp3-text-muted"
                  >
                    (必須)
                  </span>
                </label>
                <div
                  className="bp3-form-content"
                >
                  <Blueprint3.InputGroup
                    id="title"
                    large={true}
                    onChange={[Function]}
                    value=""
                  >
                    <div
                      className="bp3-input-group bp3-large"
                    >
                      <Blueprint3.Icon />
                      <input
                        className="bp3-input"
                        id="title"
                        onChange={[Function]}
                        style={
                          Object {
                            "paddingRight": 10,
                          }
                        }
                        type="text"
                        value=""
                      />
                    </div>
                  </Blueprint3.InputGroup>
                </div>
              </div>
            </Blueprint3.FormGroup>
            <Blueprint3.FormGroup
              label="概要"
              labelFor="catchMessage"
            >
              <div
                className="bp3-form-group"
              >
                <label
                  className="bp3-label"
                  htmlFor="catchMessage"
                >
                  概要
                   
                  <span
                    className="bp3-text-muted"
                  />
                </label>
                <div
                  className="bp3-form-content"
                >
                  <Blueprint3.InputGroup
                    id="catchMessage"
                    large={true}
                    onChange={[Function]}
                    value=""
                  >
                    <div
                      className="bp3-input-group bp3-large"
                    >
                      <Blueprint3.Icon />
                      <input
                        className="bp3-input"
                        id="catchMessage"
                        onChange={[Function]}
                        style={
                          Object {
                            "paddingRight": 10,
                          }
                        }
                        type="text"
                        value=""
                      />
                    </div>
                  </Blueprint3.InputGroup>
                </div>
              </div>
            </Blueprint3.FormGroup>
            <Blueprint3.FormGroup
              label="場所"
              labelFor="place"
            >
              <div
                className="bp3-form-group"
              >
                <label
                  className="bp3-label"
                  htmlFor="place"
                >
                  場所
                   
                  <span
                    className="bp3-text-muted"
                  />
                </label>
                <div
                  className="bp3-form-content"
                >
                  <Blueprint3.InputGroup
                    id="place"
                    large={true}
                    onChange={[Function]}
                    value=""
                  >
                    <div
                      className="bp3-input-group bp3-large"
                    >
                      <Blueprint3.Icon />
                      <input
                        className="bp3-input"
                        id="place"
                        onChange={[Function]}
                        style={
                          Object {
                            "paddingRight": 10,
                          }
                        }
                        type="text"
                        value=""
                      />
                    </div>
                  </Blueprint3.InputGroup>
                </div>
              </div>
            </Blueprint3.FormGroup>
            <Blueprint3.FormGroup
              label="開始日時"
              labelFor="startedAt"
            >
              <div
                className="bp3-form-group"
              >
                <label
                  className="bp3-label"
                  htmlFor="startedAt"
                >
                  開始日時
                   
                  <span
                    className="bp3-text-muted"
                  />
                </label>
                <div
                  className="bp3-form-content"
                >
                  <Blueprint3.InputGroup
                    id="startedAt"
                    large={true}
                    onChange={[Function]}
                    placeholder="2018/1/23 9:30"
                    value=""
                  >
                    <div
                      className="bp3-input-group bp3-large"
                    >
                      <Blueprint3.Icon />
                      <input
                        className="bp3-input"
                        id="startedAt"
                        onChange={[Function]}
                        placeholder="2018/1/23 9:30"
                        style={
                          Object {
                            "paddingRight": 10,
                          }
                        }
                        type="text"
                        value=""
                      />
                    </div>
                  </Blueprint3.InputGroup>
                </div>
              </div>
            </Blueprint3.FormGroup>
            <Blueprint3.FormGroup
              label="終了日時"
              labelFor="endedAt"
            >
              <div
                className="bp3-form-group"
              >
                <label
                  className="bp3-label"
                  htmlFor="endedAt"
                >
                  終了日時
                   
                  <span
                    className="bp3-text-muted"
                  />
                </label>
                <div
                  className="bp3-form-content"
                >
                  <Blueprint3.InputGroup
                    id="endedAt"
                    large={true}
                    onChange={[Function]}
                    placeholder="2018/1/23 9:30"
                    value=""
                  >
                    <div
                      className="bp3-input-group bp3-large"
                    >
                      <Blueprint3.Icon />
                      <input
                        className="bp3-input"
                        id="endedAt"
                        onChange={[Function]}
                        placeholder="2018/1/23 9:30"
                        style={
                          Object {
                            "paddingRight": 10,
                          }
                        }
                        type="text"
                        value=""
                      />
                    </div>
                  </Blueprint3.InputGroup>
                </div>
              </div>
            </Blueprint3.FormGroup>
            <Blueprint3.Button
              fill={true}
              large={true}
              onClick={[Function]}
              text="作成"
            >
              <button
                className="bp3-button bp3-fill bp3-large"
                onClick={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                type="button"
              >
                <Blueprint3.Icon
                  key="leftIcon"
                />
                <span
                  className="bp3-button-text"
                  key="text"
                >
                  作成
                </span>
                <Blueprint3.Icon
                  key="rightIcon"
                />
              </button>
            </Blueprint3.Button>
          </form>
        </EventCreateForm>
      </div>
    </StyledComponent>
  </styled.div>
</NewEvent>
`;
